collect_app/src/main/res/layout/widget_answer.xml

```diff
@@ -5,25 +5,21 @@
     android:layout_height="wrap_content">
 
     <org.odk.collect.android.views.MultiClickSafeButton
-        android:id="@+id/url_button"
+        android:id="@+id/widget_button"
         style="@style/Widget.Collect.Button.Custom"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:padding="@dimen/margin_small"
-        android:text="@string/open_url"
-        android:layout_marginStart="@dimen/margin_standard"
-        android:layout_marginEnd="@dimen/margin_standard"
-        android:layout_marginTop="@dimen/margin_extra_small">
+        android:layout_marginHorizontal="@dimen/margin_standard"
```

Comment:
Instead of setting `android:layout_marginHorizontal="@dimen/margin_standard"` here and below it would be better to set hroizontalPadding just once in the parent LinearLayout.
I also think that the parent LinearLayout should have some verticalPadding margin_small or margin_extra_small because otherwise those widgets might be too close to each other if you use field-list.
Have you tested such a scenario? If so please attach a screenshot.

Reply:
Currently, I'm using `widget_answer.xml` for URL widget and Date and Time widgets. You were correct, there was some spacing issue in case of multiple widgets on the same screen. I added `vertical padding` as `margin_small`.

| On master  | On Rework Date Widget branch |
| ------------- | ------------- |
| ![WhatsApp Image 2020-07-30 at 03 01 15 (1)](https://user-images.githubusercontent.com/35730054/88855868-27122900-d211-11ea-9f22-ee09a7dccf3d.jpeg)  | ![WhatsApp Image 2020-07-30 at 03 01 15](https://user-images.githubusercontent.com/35730054/88855873-29748300-d211-11ea-9292-6fca48958a62.jpeg) |

Reply:
Yes I think like for now it's ok. I guess once you finish reworking all the widgets we should go through all layouts and make sure they are consistent because we might miss something (you changing the code and me reviewing it).